scipy/ndimage/morphology.py

```diff
@@ -1878,9 +1879,11 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
     chessboard algorithms.
 
     """
-    if (not return_distances) and (not return_indices):
-        msg = 'at least one of distances/indices must be specified'
-        raise RuntimeError(msg)
+    ft_inplace = isinstance(indices, numpy.ndarray)
+    dt_inplace = isinstance(distances, numpy.ndarray)
```

Comment:
We definitely do not test any `ndarray` subclasses currently, so I wouldn't say it needs to be supported, but it did work, at least for `xarray`. 

Example:

```Python
import numpy as np
import xarray
import scipy.ndimage as ndi

img = np.zeros((16, 16))
img[4:-4, 4:-4] = 1
img = xarray.DataArray(img)
out = xarray.DataArray(np.empty((16, 16)))

ndi.distance_transform_bf(img, return_distances=True, return_indices=False, distances=out)
print(out)

```

In the example above, `return` does end up including the distances as well since the code prior to this PR also had `isinstance` checks in another location based on `numpy.ndarray`, but after this PR, an error would be raised instead.

Reply:
I see. I just copied that from the other two functions since it was already there.